### PR TITLE
Expense notification cleanup

### PR DESCRIPTION
**Pull request by asood123**

### CODE REVIEW

**Comment by asood123** [comment]
Details on the changes here: https://github.com/OpenCollective/OpenCollective/issues/281

**Comment by asood123** [comment]
This improves expense notifications. Good enough to prevent ugly emails from going out.

File: server/constants/activities.js [hunk 1]
```diff
@@ -15,13 +15,16 @@ export default {
...
   GROUP_USER_ADDED: 'group.user.added',
   SUBSCRIPTION_CONFIRMED: 'subscription.confirmed',
   SUBSCRIPTION_CANCELED: 'subscription.canceled',
   USER_CREATED: 'user.created',
   WEBHOOK_STRIPE_RECEIVED: 'webhook.stripe.received',
   WEBHOOK_PAYPAL_RECEIVED: 'webhook.paypal.received',
-  GROUP_MONTHLY: 'group.monthly'
+  GROUP_MONTHLY: 'group.monthly',
+
+  // Not used anymore, leaving for historical reference
```
Comment (xdamman) [inline server/constants/activities.js]: Why do you want to keep this for historical reason?
Don't worry, Git never forgets ;-)

File: server/lib/email.js [hunk 2]
```diff
@@ -241,6 +241,11 @@ const sendMessageFromActivity = (activity, notification) => {
         reject: notification.User.generateLoginLink(`/${data.group.slug}/expenses/${data.expense.id}/reject`)
       };
       return generateEmailFromTemplateAndSend('group.expense.created', notification.User.email, data);
+    case activities.GROUP_EXPENSE_APPROVED:
+      data.actions = {
+        view: notification.User.generateLoginLink(`/${data.group.slug}/transactions/expenses#exp${data.expense.id}`)
```
Comment (xdamman) [inline server/lib/email.js]: Can you name this viewExpenseUrl?
A view is a different thing.

File: server/lib/email.js [hunk 3]
```diff
@@ -241,6 +241,11 @@ const sendMessageFromActivity = (activity, notification) => {
...
+      }
+      return generateEmailFromTemplateAndSend('group.expense.approved.for.host', notification.User.email, data);
```
Comment (xdamman) [inline server/lib/email.js]: Why "for.host"? Is there another similar template that is not for host? Also, does it matter if it's for someone else?

File: templates/emails/group.expense.approved.for.host.hbs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+.btn.red {
+  background: #dd2663;
+}
+</style>
+
+<center>
+  <h1>{{{currency expense.amount currency=expense.currency}}}</h1>
+  <div>{{expense.title}}</div>
+  <p><font color="#999">{{expense.notes}}</font></p>
```
Comment (xdamman) [inline templates/emails/group.expense.approved.for.host.hbs]: Would be good to show it again who filed the expense originally and who approved it.

File: templates/emails/group.expense.paid.hbs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  vertical-align: middle;
+}
+
+.btn.red {
+  background: #dd2663;
+}
+</style>
+
+<center>
+  <h3> Hi there! Just a note to let you know that an expense was paid in {{group.name}}. </h3>
```
Comment (xdamman) [inline templates/emails/group.expense.paid.hbs]: Would be good to be able to differentiate the filer to say "your expense"

File: server/lib/email.js (hunk 3 above)
Comment (asood123) [inline server/lib/email.js]: Yes, we need another template here to send an email to the members and the expense filer letting them know that the expense is approved.

File: templates/emails/group.expense.approved.for.host.hbs (hunk 4 above)
Comment (asood123) [inline templates/emails/group.expense.approved.for.host.hbs]: yep, it's on the list. Getting who approved it is a bit more work because you have to fetch that user info. Let's do it in a different PR.

File: templates/emails/group.expense.paid.hbs (hunk 5 above)
Comment (asood123) [inline templates/emails/group.expense.paid.hbs]: That's the eventual goal. That requires different templates - we'll do it in another PR. Let's get this merged to fix current emails with broken images and missing subject.

File: server/constants/activities.js (hunk 1 above)
Comment (asood123) [inline server/constants/activities.js]: When a new engineer sees this in the database, it's much easier to reference it in the codebase, than try to dig through git history.
